### PR TITLE
Optimise integration tests

### PR DESCRIPTION
**Pull request by ikbalkaya**
This PR improves Integration test performance by reusing some objects for all tests

### CODE REVIEW

File: src/test/java/com/ably/kafka/connect/integration/ChannelSinkTaskTest.java [hunk 1]
```diff
@@ -5,24 +5,30 @@
...
 import java.util.Map;
+import java.util.concurrent.CountDownLatch;
 
 import static org.apache.kafka.connect.runtime.ConnectorConfig.CONNECTOR_CLASS_CONFIG;
 import static org.apache.kafka.connect.runtime.ConnectorConfig.KEY_CONVERTER_CLASS_CONFIG;
 import static org.apache.kafka.connect.runtime.ConnectorConfig.TASKS_MAX_CONFIG;
 import static org.apache.kafka.connect.runtime.ConnectorConfig.VALUE_CONVERTER_CLASS_CONFIG;
 import static org.apache.kafka.connect.runtime.SinkConnectorConfig.TOPICS_CONFIG;
-import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (KacperKluka) [inline src/test/java/com/ably/kafka/connect/integration/ChannelSinkTaskTest.java:31]: It looks that the IDE did it again and used the `*` import :sweat_smile:

File: src/test/java/com/ably/kafka/connect/AblyHelpers.java [hunk 2]
```diff
@@ -72,7 +72,7 @@ public String key() {
     public static AppSpec createTestApp() throws AblyException {
         ClientOptions opts = new ClientOptions("a.b:c");
         opts.environment = TEST_ENVIRONMENT;
-        AblyRest ably = new AblyRest(opts);
+        AblyRealtime ably = new AblyRealtime(opts);
```
Comment (KacperKluka) [inline src/test/java/com/ably/kafka/connect/AblyHelpers.java:75]: What's the purpose of this change in terms of optimizing tests? :thinking:

File: src/test/java/com/ably/kafka/connect/integration/ChannelSinkTaskTest.java [hunk 3]
```diff
@@ -46,29 +53,38 @@
...
+        ablyClient.connection.on(connectionStateChange -> {
+            if (connectionStateChange.current == ConnectionState.connected) {
+                latch.countDown();
+            } else if (connectionStateChange.current == ConnectionState.failed) {
+                latch.countDown();
+                assertTrue(false, "Connection failed");
+            }
+        });
+        latch.await();
     }
```
Comment (KacperKluka) [inline src/test/java/com/ably/kafka/connect/integration/ChannelSinkTaskTest.java:75]: I'd extract this part to a separate method called something like `assertAblyIsConnected()` to make it clear what are we trying to do here :wink: 
```suggestion
        ablyClient = AblyHelpers.realtimeClient(appSpec);
        assertAblyIsConnected();
    }

    private void assertAblyIsConnected() {
        final CountDownLatch latch = new CountDownLatch(1);
        ablyClient.connection.on(connectionStateChange -> {
            if (connectionStateChange.current == ConnectionState.connected) {
                latch.countDown();
            } else if (connectionStateChange.current == ConnectionState.failed) {
                latch.countDown();
                assertTrue(false, "Connection failed");
            }
        });
        latch.await();
    }
```

File: src/test/java/com/ably/kafka/connect/integration/ChannelSinkTaskTest.java [hunk 4]
```diff
@@ -82,7 +98,7 @@ public void testMessagePublish_correctConfigAtLeastATaskIsRunning() throws Excep
     public void testMessagePublish_channelExistsWithStaticChannelName() {
         final String channelName = "test-channel";
 
-        connectCluster.kafka().createTopic(DEFAULT_TOPIC);
+
```
Comment (KacperKluka) [inline src/test/java/com/ably/kafka/connect/integration/ChannelSinkTaskTest.java:101]: Please remove this and all the other blank lines added in this PR :bow:

**Review by KacperKluka — CHANGES_REQUESTED** [review]
Looks good but some minor improvements can be made :wink:

File: src/test/java/com/ably/kafka/connect/AblyHelpers.java (hunk 2 above)
Comment (ikbalkaya) [reply src/test/java/com/ably/kafka/connect/AblyHelpers.java:75]: I found out it was faster to use AblyRealtime + I didn't find reason why we are using AblyRest 🤷

File: src/test/java/com/ably/kafka/connect/integration/ChannelSinkTaskTest.java (hunk 3 above)
Comment (ikbalkaya) [reply src/test/java/com/ably/kafka/connect/integration/ChannelSinkTaskTest.java:80]: a6c110fd320ce329dedc8bbdc1956d6c750f7a83

File: src/test/java/com/ably/kafka/connect/integration/ChannelSinkTaskTest.java (hunk 1 above)
Comment (ikbalkaya) [reply src/test/java/com/ably/kafka/connect/integration/ChannelSinkTaskTest.java]: 9395af0d1cfb6fd65ea70ce427513c812196c596

File: src/test/java/com/ably/kafka/connect/integration/ChannelSinkTaskTest.java (hunk 4 above)
Comment (ikbalkaya) [reply src/test/java/com/ably/kafka/connect/integration/ChannelSinkTaskTest.java]: https://github.com/ably/kafka-connect-ably/pull/61/commits/6ee1a15d5f2ce5ce8c190ec6e57c0e267e292d00

Comment (ikbalkaya) [reply src/test/java/com/ably/kafka/connect/integration/ChannelSinkTaskTest.java]: ea8bcf079267c4f96b607f2b125fff92c7d15456
